### ISSUE TITLE
after git calls, make RELEASE=1 fails on stock container, x86_64 environment

### ISSUE THREAD

**Issue by ChristinMilloy** [DirtyHairy/r77-firmware-ng-build#1]
Followed git instructions provided, but `make RELEASE=1` terminates with errors after a long build process.
Build output file attached below. Also here's output of `lscpu`:

`Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   39 bits physical, 48 bits virtual
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              1
Core(s) per socket:              4
Socket(s):                       1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           95
Model name:                      Intel(R) Atom(TM) CPU C3538 @ 2.10GHz
Stepping:                        1
CPU MHz:                         2100.000
CPU max MHz:                     2100.0000
CPU min MHz:                     800.0000
BogoMIPS:                        4199.97
Virtualization:                  VT-x
L1d cache:                       96 KiB
L1i cache:                       128 KiB
L2 cache:                        8 MiB
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:        Vulnerable, IBPB: disabled, STIBP: disabled
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm consta
                                 nt_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg cx16 xtpr pdcm sse4_
                                 1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave rdrand lahf_lm 3dnowprefetch epb intel_pt ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept v
                                 pid fsgsbase tsc_adjust smep erms mpx rdseed smap clflushopt sha_ni xsaveopt xsavec xgetbv1 dtherm arat pln pts md_clear arch_capabilities`

[output.txt](https://github.com/DirtyHairy/r77-firmware-ng-build/files/9441236/output.txt)

**Comment by ChristinMilloy** [comment]
Regular `make` for development image produces same result (and no uImage generated either way)

**Comment by ChristinMilloy** [comment]
So the problem is definitely in compiling stella. I started by pulling latest stella master into the stella folder, and that started generating different errors relating to paths for src not being included properly. I'm just filling in the missing relative paths manually for now in the stella source code, but I suspect they might have some kind of reference errors (or missing PATH) when compiling stella with the retron77 flags set. I don't know enough about C++ to find and fix whatever that is, but hard-coding relative paths where the compiler finds reference errors seems to be working... I'll follow up if I can finally get a successful build or not.

**Comment by ChristinMilloy** [comment]
No dice. A time consuming effort, replacing broken references with relative paths, ended in failure. getting errors in sqlite compile that I don't know how to fix. Either I'm doing something really wrong, or stella's code isn't compatible with this build container. :(

**Comment by DirtyHairy** [comment]
Sorry, I had a local change that I hadn't commited. Committed now; start over with a new container, do `git pull && git update submodules --recursive` and the build should go through.

The R77 does not use Stella's standard `configure && make`, so adjustments need to be made to the build config from time to time.

**Comment by ChristinMilloy** [comment]
Awesome :)

So first I destroyed the container and image and re-cloned the image (then I realized I didn't actually need to do that -- oops). Did `git pull` then `git submodule update --init --recursive`

You are a genius. With your new commit, `make RELEASE=1` completed perfectly. It does take an awfully long time in the container, but that's a small price to pay. I'll report back on the forum if I'm able to get my network mount thing working.

Thank you so much!
